### PR TITLE
[Rules] Reorganized folders and files

### PR DESCRIPTION
**Pull request by fusionrace**
**What**
- Added the following top level folder categories:
-- apt
-- generic
--hacktool
--malware
--ransomware

- These folders are then broken down the following operating systems:
-- linux
-- macos
-- multi (multiple operating systems affected)
-- windows

- All yara rules have now been moved to their own individual files within the respective operating system folders

**Why**
- This update improves the ability to understand where gaps are by operating systems and by category type.   
- Easier to manage 

**Testing**
- Locally and waiting on CI before I seek approval.

### CODE REVIEW

**Comment by fusionrace** [comment]
@mime-frame @austinbyers PTAL the updated folder structure and advise of any changes.  Thanks!

**Comment by mime-frame** [comment]
@fusionrace lets get rid of the `generic` folder, since it's generic, there's 0 rules in it and I'm not sure how we'd quite use it. I'm also thinking we remove the `apt` folder since APT's use both private and public tools and it's easy to get confused. Ex: `rules/public/malware/windows/malware_windows_moonlightmaze_cle_tool.yara` - MoonlightMaze is APT.

File: rules/public/malware/macos/malware_macos_proton_rat_generic.yara [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    condition:
+        uint32(0) == 0xfeedface or uint32(0) == 0xcefaedfe or uint32(0) == 0xfeedfacf or uint32(0) == 0xcffaedfe or uint32(0) == 0xcafebabe or uint32(0) == 0xbebafeca
+}
+
+rule malware_macos_proton_rat_generic
+{
+    meta:
+        description = "https://www.hackread.com/hackers-selling-undetectable-proton-mac-malware/"
+        reference = "https://objective-see.com/blog/blog_0x1D.html"
+        author = "Daimon Greaves"
```
Comment (mime-frame) [inline rules/public/malware/macos/malware_macos_proton_rat_generic.yara]: i wrote this one :)

File: rules/public/ransomware/windows/ransomware_windoes_powerware_locky.yara [hunk 2]
```diff
@@ -0,0 +1,17 @@
+rule ransomware_windoes_powerware_locky
```
Comment (mime-frame) [inline rules/public/ransomware/windows/ransomware_windoes_powerware_locky.yara]: `windows`

File: rules/public/hacktool/multi/hacktool_multi_masscan.yara [hunk 3]
```diff
@@ -0,0 +1,17 @@
+rule hacktool_multi_masscan
+{
+    meta:
+        description = "masscan is a performant port scanner, it produces results similar to nmap"
+        reference = "https://github.com/robertdavidgraham/masscan"
+        author = "Airbnb CSIRT"
```
Comment (mime-frame) [inline rules/public/hacktool/multi/hacktool_multi_masscan.yara]: while you're in here, can you change all of the `Airbnb CSIRT` to `@mimeframe`?

File: rules/public/malware/macos/malware_macos_proton_rat_generic.yara (hunk 1 above)
Comment (fusionrace) [inline rules/public/malware/macos/malware_macos_proton_rat_generic.yara]: Sorry about that =)

File: rules/public/hacktool/multi/hacktool_multi_masscan.yara (hunk 3 above)
Comment (fusionrace) [inline rules/public/hacktool/multi/hacktool_multi_masscan.yara]: Sounds good

File: rules/public/ransomware/windows/ransomware_windoes_powerware_locky.yara (hunk 2 above)
Comment (fusionrace) [inline rules/public/ransomware/windows/ransomware_windoes_powerware_locky.yara]: Done.

File: rules/public/malware/macos/malware_macos_proton_rat_generic.yara [hunk 4]
```diff
@@ -0,0 +1,27 @@
+private rule MachO
+{
+    meta:
+        description = "Mach-O binaries"
+    condition:
+        uint32(0) == 0xfeedface or uint32(0) == 0xcefaedfe or uint32(0) == 0xfeedfacf or uint32(0) == 0xcffaedfe or uint32(0) == 0xcafebabe or uint32(0) == 0xbebafeca
+}
```
Comment (austinbyers) [inline rules/public/malware/macos/malware_macos_proton_rat_generic.yara]: The `MachO` rule should be moved into its own file and included by each individual rules file.

`macho.yara`:  `private rule MachO { ... }`

`malware_macos_proton_rat_generic.yara`:
```
include "../macho.yara"

....
```

See http://yara.readthedocs.io/en/v3.6.3/writingrules.html#including-files

**Comment by fusionrace** [comment]
Working on the path issue.

**Comment by fusionrace** [comment]
@austinbyers finally... figured that out.  Please approve when ready.  Thanks!

**Pull request merged**